poweriq-worker/src/main/java/com/vmware/flowgate/poweriqworker/jobs/PowerIQService.java

```diff
@@ -1302,30 +1305,55 @@ public PowerIQAPIClient createClient(FacilitySoftwareConfig poweriq) {
             }
             String rate_current = pduInfoFromPowerIQ.get(FlowgateConstant.PDU_RATE_AMPS);
             DecimalFormat df = new DecimalFormat("#.00");
+            if(totalOutletCurrentUsed == 0.0) {
+               totalOutletCurrentUsed = inletTotalCurrent;
+            }
             if(rate_current != null) {
                Double rate_current_value = Double.parseDouble(rate_current);
                ValueUnit current_load = new ValueUnit();
                current_load.setKey(MetricName.PDU_CURRENT_LOAD);
                current_load.setTime(valueTime);
                current_load.setUnit(RealtimeDataUnit.Percent.toString());
-               current_load.setValueNum(Double.parseDouble(df.format(totalCurrentUsed/rate_current_value)));
+               current_load.setValueNum(Double.parseDouble(df.format(totalOutletCurrentUsed/rate_current_value)));
                values.add(current_load);
             }
-            String rate_power = pduInfoFromPowerIQ.get(FlowgateConstant.PDU_MIN_RATE_POWER);
-            if(rate_power != null) {
-               Double rate_power_value = Double.parseDouble(rate_power);
+            String real_rate_power = getRealRatePower(inletVoltage, pduInfoFromPowerIQ);
+            if(totalOutletPowerUsed == 0.0) {
+               totalOutletPowerUsed = inlteTotalPower;
+            }
+            if(real_rate_power != null) {
+               Double rate_power_value = Double.parseDouble(real_rate_power);
                ValueUnit power_load = new ValueUnit();
                power_load.setKey(MetricName.PDU_POWER_LOAD);
                power_load.setTime(valueTime);
                power_load.setUnit(RealtimeDataUnit.Percent.toString());
-               power_load.setValueNum(Double.parseDouble(df.format(totalPowerUsed/rate_power_value)));
+               power_load.setValueNum(Double.parseDouble(df.format(totalOutletPowerUsed/rate_power_value)));
                values.add(power_load);
             }
          }
       }
       return values;
    }
 
+   public String getRealRatePower(double inletVoltage, Map<String,String> pduInfoFromPowerIQ) {
+      String min_rate_voltage =  pduInfoFromPowerIQ.get(FlowgateConstant.PDU_MIN_RATE_VOLTS);
+      String max_rate_voltage =  pduInfoFromPowerIQ.get(FlowgateConstant.PDU_MAX_RATE_VOLTS);
+      double absoluteValueForMinVolt = 0.0;
+      double absoluteValueForMaxVolt = 0.0;
+      if(min_rate_voltage != null) {
+         Double min_rate_voltage_value = Double.parseDouble(min_rate_voltage);
+         absoluteValueForMinVolt = Math.abs(inletVoltage - min_rate_voltage_value);
+      }
+      if(max_rate_voltage != null) {
+         Double max_rate_voltage_value = Double.parseDouble(max_rate_voltage);
+         absoluteValueForMaxVolt = Math.abs(inletVoltage - max_rate_voltage_value);
+      }
+      if(absoluteValueForMinVolt < absoluteValueForMaxVolt) {
+         return pduInfoFromPowerIQ.get(FlowgateConstant.PDU_MIN_RATE_POWER);
```

Comment:
as you already define the variable in the beginning you don't need to read some thing from the map again,

also you should return a double instead of string. since later you will need to convert it from string to double anyway.